### DISCUSSION TITLE
Tauri action not building Windows executable

### DISCUSSION THREAD

**Discussion by opeolluwa** [opeolluwa/filesync#155] [General]
![Screenshot from 2023-06-23 06-32-29_063359](https://github.com/opeolluwa/send-file/assets/67811147/de63caee-7934-4f82-bfef-dd8b5536c4bb)
 The following error is noticed in building the application for Windows Operating system.

**Comment by opeolluwa** [comment]
The error is due to a system requirement for one of the crates used 


pnet_datalink on windows here:  https://crates.io/crates/pnet_datalink

**Comment by opeolluwa** [comment]
![Screenshot_20230624-224226](https://github.com/opeolluwa/send-file/assets/67811147/5bc37466-a28b-4656-b231-3663ce4f54c7)
https://crates.io/crates/pnet_datalink

**Reply by opeolluwa** [reply]
Check out the link

**Reply by opeolluwa** [reply]
If there are any errors along the way, please leave a comment

**Comment by Dev-wonderful** [comment]
Hello, I followed the instructions highlighted there, but I'm still getting the same error, particularly because I don't understand the last instruction. To put `packet.lib` inside of the `lib` directory in `winpcap` developers pack, into the root of `this repository`, which repo are they referring to? Also, `winpcap` advised us to download `npcap`, because they their repo has been deprecated, and `npcap` automatically has support for `winpcap`. The difference is, there is no `packet.lib`, what we have is `packet.dll`. 

A screenshot of the requirement is below

**Reply by Dev-wonderful** [reply]
![windows requirement](https://github.com/opeolluwa/send-file/assets/69244197/c96c1833-8e69-43b3-8cdd-f86544303906)

**Reply by Dev-wonderful** [reply]
UPDATE: I didn't realize that `WinPcap` is different from `WinPcap Developers pack`. I was able to download the developers pack and find the `Packet.lib` file. I used the one inside of x64 directory.

For the final part, placing the `Packet.lib` in the root of the repository, is also correct but in the context of this send-file repo, you would have to place it inside of the core directory in this send-file repo.

**Reply by opeolluwa** [reply]
Did it work?

**Reply by Dev-wonderful** [reply]
Yes it did, but it took 10 mins to compile, I don't know if it's normal

**Comment by opeolluwa** [comment]
Also, try to sync the fork and run the project, I rewrote the section using [target specific API](https://github.com/opeolluwa/send-file/blob/master/core/src/net/ip_manager.rs#L30) However,  I haven't been able to try out

**Reply by Dev-wonderful** [reply]
Yeah I synced the fork, about an hour ago, except if you've updated it again.

I noticed this comment or error after compiling `thread 'tokio-runtime-worker' panicked at 'No Ip address detected: ()', src\server\mod.rs:61:10`, also when I clicked on settings, the app hanged.

**Reply by opeolluwa** [reply]
I updated the build config some minutes ago.
Regards the compile time, I think it's fair enough, it will reduce on subsequent runs.

**Comment by opeolluwa** [comment]
Could you checkout a new branch and run the application without the package.lib, let's verify if it works

**Reply by Dev-wonderful** [reply]
Do you want me to update my fork with your new build config, then create a new branch on my fork without the packet.lib or you already have a branch you want me to checkout?

**Reply by opeolluwa** [reply]
No, I don't have a branch. You'll have to create one

**Reply by Dev-wonderful** [reply]
The UI code has an error, so it's breaking the dev build, import error and tsconfig issues

**Reply by Dev-wonderful** [reply]
> The UI code has an error, so it's breaking the dev build, import error and tsconfig issues

![IMG_20230629_154730_011.jpg](https://github.com/opeolluwa/send-file/assets/69244197/78e0da2e-d452-4670-83a2-ca8c1c17fe6c)

**Reply by Dev-wonderful** [reply]
Sorry that I couldn't use a screenshot

**Reply by opeolluwa** [reply]
It's fine.

**Comment by opeolluwa** [comment]
Check your copy of the repo if the script section package.json has a "preinstall" command. If not, please, pull recent update and run yarn install, that should fix it

Otherwise, enter the browser directory and run `yarn install`
